cms/models.py

```diff
@@ -5,13 +5,59 @@
 
 from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
 from wagtail.core import blocks
-from wagtail.core.models import Page
+from wagtail.core.models import Page, Orderable
 from wagtail.core.fields import RichTextField, StreamField
 from wagtail.core.blocks import RawHTMLBlock
 from wagtail.images.models import Image
 from wagtail.images.blocks import ImageChooserBlock
 
 
+class CourseProgramChildPage(Page):
+    """
+    Abstract page representing a child of Course/Program Page
+    """
+    class Meta:
+        abstract = True
+
+    parent_page_types = ['CoursePage', 'ProgramPage']
+
+    @classmethod
+    def can_create_at(cls, parent):
+        # You can only create one of these page under course / program.
+        return super(CourseProgramChildPage, cls).can_create_at(parent) \
+               and parent.get_children().type(cls).count() == 0
+
+
+class LearningOutComesPage(CourseProgramChildPage):
```

Comment:
Could you rename it to `LearningOutcomesPage`? (small c in come), otherwise it will get rendered as `Learning Out Comes Page` in the admin description.